core/src/main/scala/org/http4s/Query.scala

```diff
@@ -160,17 +179,20 @@ object Query {
     * If parsing fails, the empty [[Query]] is returned
     */
   def fromString(query: String): Query =
-    if (query.isEmpty) new Query(Vector("" -> None))
-    else
-      QueryParser.parseQueryString(query) match {
-        case Right(query) => query
-        case Left(_) => Query.empty
-      }
+    new Query(Right(query))
```

Comment:
Actually, this allows invalid queries.  We need to make sure it's all pchars.

We foolishly made this return a `Query` instead of a `ParseResult[Query]`.  We can't fix that now, but can in 1.0.  I guess for compatibility, we should return the empty string if it's invalid. :vomiting_face: